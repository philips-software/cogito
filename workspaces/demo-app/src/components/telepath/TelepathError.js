import React from 'react'
import { TimedStatus } from 'components/utils'
import { StatusSegmentRow } from 'components/ui/layout'

export const TelepathError = ({ children, error, onTimeout, timeout = 3000 }) => {
  if (!error && !children) {
    return null
  }
  return (
    <TimedStatus timeout={timeout} onTimeout={onTimeout}>
      <StatusSegmentRow>{error || children}</StatusSegmentRow>
    </TimedStatus>
  )
}
