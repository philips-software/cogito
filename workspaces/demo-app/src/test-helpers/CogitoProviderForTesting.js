class CogitoProviderForTesting {
  constructor ({ originalCogitoProvider, redirectProvider }) {
    this.cogitoProvider = originalCogitoProvider
    this.redirectProvider = redirectProvider
  }
  send (payload, callback) {
    if (payload.method === 'eth_accounts' ||
        payload.method === 'eth_sendTransaction') {
      this.redirectProvider.send(payload, callback)
    } else {
      this.cogitoProvider.send(payload, callback)
    }
  }

  sendAsync () {
    this.send(...arguments)
  }
}

export { CogitoProviderForTesting }
