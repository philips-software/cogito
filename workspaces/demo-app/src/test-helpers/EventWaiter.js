class EventWaiter {
  promise
  promiseResolve
  promiseReject
  expectedNumberOfEvents = 1
  intermediateValues = []

  constructor () {
    this.promise = new Promise((resolve, reject) => {
      this.promiseResolve = resolve
      this.promiseReject = reject
    })
  }

  reset () {
    this.promise = new Promise((resolve, reject) => {
      this.promiseResolve = resolve
      this.promiseReject = reject
    })
    this.expectedNumberOfEvents = 1
  }

  onValueChanged = value => {
    if (--this.expectedNumberOfEvents === 0) {
      this.promiseResolve(value)
      clearTimeout(this.timeout)
    } else {
      this.intermediateValues = [ ...this.intermediateValues, value ]
    }
  }

  expect (expectedNumberOfEvents) {
    this.expectedNumberOfEvents = expectedNumberOfEvents
  }

  wait () {
    this.timeout = setTimeout(() => {
      this.promiseReject(new Error('timedout'))
    }, 4000)
    return this.promise
  }
}

export { EventWaiter }
