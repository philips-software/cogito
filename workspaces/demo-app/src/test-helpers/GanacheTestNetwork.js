import Web3 from 'web3'
import ganache from 'ganache-cli'
import initContract from 'truffle-contract'

class GanacheTestNetwork {
  mnemonic = 'hair snack volcano shift tragic wrong wreck release vibrant gossip ugly debate'
  _web3

  constructor () {
    const provider = ganache.provider({
      mnemonic: this.mnemonic
    })
    this._web3 = new Web3(provider)
  }

  get web3 () {
    return this._web3
  }

  getAccounts = async () => {
    this.accounts = await this.web3.eth.getAccounts()
    return this.accounts
  }

  deploy = async (contractDefinition, { from }) => {
    const SimpleStorage = new this.web3.eth.Contract([contractDefinition])
    const deployedContract = await SimpleStorage.deploy({
      data: contractDefinition.bytecode
    }).send({
      from,
      gas: 1000000,
      gasPrice: '10000000000000'
    })
    const contract = initContract(contractDefinition)
    contract.setProvider(this.web3.currentProvider)

    return contract.at(deployedContract.options.address)
  }
}

export { GanacheTestNetwork }
