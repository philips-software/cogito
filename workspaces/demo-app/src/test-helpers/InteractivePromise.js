class InteractivePromise {
  promise
  resolve
  reject
  constructor () {
    this.promise = new Promise((resolve, reject) => {
      this.resolve = resolve
      this.reject = reject
    })
  }
  resolve (value) {
    this.resolve(value)
  }
  reject (value) {
    this.reject(value)
  }
  get () {
    return this.promise
  }
}

export { InteractivePromise }
