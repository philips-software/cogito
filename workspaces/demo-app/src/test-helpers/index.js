export { inRouter, renderInRouter } from './router'
export { EventWaiter } from './EventWaiter'
export { GanacheTestNetwork } from './GanacheTestNetwork'
