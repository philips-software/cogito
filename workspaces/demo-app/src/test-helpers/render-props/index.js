export {
  renderWithStore as render,
  rerenderWithStore as rerender
} from './renderWithStore'
export * from 'react-testing-library'
