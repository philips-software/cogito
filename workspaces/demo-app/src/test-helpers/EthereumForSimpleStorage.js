import { SimpleStorage } from '@cogitojs/demo-app-contracts'
import { CogitoEthereum } from '@cogitojs/cogito'
import { GanacheTestNetwork } from './GanacheTestNetwork'

class EthereumForSimpleStorage {
  ganacheTestNetwork
  context

  injectWeb3 = () => {
    window.web3 = this.ganacheTestNetwork.web3
    process.env.REACT_APP_USE_INJECTED_WEB3 = 'YES'
  }

  get address () {
    return this.ganacheTestNetwork.wallet.address
  }

  get telepathChannel () {
    return this.context.telepathChannel
  }

  get cogitoWeb3 () {
    return this.context.cogitoWeb3
  }

  get simpleStorageProxy () {
    return this.context.contractsProxies.SimpleStorage
  }

  constructor ({ appName }) {
    this.appName = appName
    this.ganacheTestNetwork = new GanacheTestNetwork()
    this.injectWeb3()
  }

  deploy = () => {
    return this.ganacheTestNetwork.deploy(SimpleStorage, { from: this.address })
  }

  setupCogitoContext = async contractBlob => {
    const cogitoEthereum = new CogitoEthereum([ contractBlob ])
    this.context = await cogitoEthereum.getContext({ appName: this.appName })

    this.ganacheTestNetwork.mockTelepathChannel(this.context.telepathChannel)
  }

  setup = async () => {
    const { deployedJSON } = await this.deploy()
    await this.setupCogitoContext(deployedJSON)
  }
}

export { EthereumForSimpleStorage }
