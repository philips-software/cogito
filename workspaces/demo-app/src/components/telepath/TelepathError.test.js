import React from 'react'
import { render } from 'react-testing-library'
import { TelepathError } from './TelepathError'
import { EventWaiter } from 'test-helpers'

describe('TelepathError', () => {
  const errorDescription = 'error description'
  it('displays provied error description', () => {
    const { getByText } = render(<TelepathError error={errorDescription} />)
    expect(getByText(errorDescription)).toBeInTheDocument()
  })

  it('also accepts error description as a child text node', () => {
    const { getByText } = render(
      <TelepathError>{errorDescription}</TelepathError>
    )
    expect(getByText(errorDescription)).toBeInTheDocument()
  })

  it('does not render anything if no error description has been provided', () => {
    const { container } = render(<TelepathError />)
    expect(container.firstChild).toBeNull()
  })

  it('fades out after provided timeout (default 3000ms)', async () => {
    expect.assertions(2)
    const eventWaiter = new EventWaiter()
    const { container } = render(
      <TelepathError error={errorDescription}
        timeout={1}
        onTimeout={() => eventWaiter.onValueChanged('timeout')}
      />
    )
    expect(await eventWaiter.wait()).toBe('timeout')
    expect(container.firstChild).toBeNull()
  })

  it('fades out after provided timeout (default 3000ms) for error provided as child node', async () => {
    expect.assertions(2)
    const eventWaiter = new EventWaiter()
    const { container } = render(
      <TelepathError timeout={1}
        onTimeout={() => eventWaiter.onValueChanged('timeout')}
      >
        {errorDescription}
      </TelepathError>
    )
    expect(await eventWaiter.wait()).toBe('timeout')
    expect(container.firstChild).toBeNull()
  })

  it('receives proper styling when error description is rendered', () => {
    const { container } = render(
      <TelepathError>{errorDescription}</TelepathError>
    )
    expect(container).toMatchSnapshot()
  })

  it('receives proper styling when error description (child) is rendered', () => {
    const { container } = render(
      <TelepathError error={errorDescription} />
    )
    expect(container).toMatchSnapshot()
  })
})
