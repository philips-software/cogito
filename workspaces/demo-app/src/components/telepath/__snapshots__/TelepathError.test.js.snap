// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelepathError receives proper styling when error description (child) is rendered 1`] = `
.css-0,
[data-css-0] {
  font-size: 0.8em;
  color: #fd81c2;
}

<div>
  <div
    style="margin: 10px; padding: 0px; box-sizing: border-box;"
  >
    <div
      class="css-ca9d80"
    >
      <div
        class="ui segment"
      >
        <p
          class="css-0"
        >
          error description
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`TelepathError receives proper styling when error description is rendered 1`] = `
.css-0,
[data-css-0] {
  font-size: 0.8em;
  color: #fd81c2;
}

<div>
  <div
    style="margin: 10px; padding: 0px; box-sizing: border-box;"
  >
    <div
      class="css-ca9d80"
    >
      <div
        class="ui segment"
      >
        <p
          class="css-0"
        >
          error description
        </p>
      </div>
    </div>
  </div>
</div>
`;
